lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -76,7 +79,9 @@ private VortexDriver(final EvaluatorRequestor evaluatorRequestor,
                        @Parameter(VortexMasterConf.WorkerMem.class) final int workerMem,
                        @Parameter(VortexMasterConf.WorkerNum.class) final int workerNum,
                        @Parameter(VortexMasterConf.WorkerCores.class) final int workerCores,
-                       @Parameter(VortexMasterConf.NumberOfVortexStartThreads.class) final int numOfStartThreads) {
+                       @Parameter(VortexMasterConf.NumberOfVortexStartThreads.class) final int numOfStartThreads,
+                       @Parameter(VortexMasterConf.InputCodec.class) final Codec inputCodec,
```

Comment:
@bgchun Thanks for the suggestion/question.
1) I think `TInput extends Codec` looks much cleaner, but given a serialized message, how could we get the codec to deserialize the input/output with? Because the codec is the part of input/output itself, I'm not sure how we can extract the codec from a byte array. Am I missing something?
2) I don't think `Kryo` will make things much easier. Similarly, we need to somehow register the input/output codec(`Serializer`). Moreover, codecs also should be sent across the wire to serialize/deserialize on Workers similar to how I made in this PR I believe.